py-polars/src/file.rs

```diff
@@ -231,7 +231,7 @@ fn get_either_file_and_path(
                     .ok()
             })
             .flatten()
-            .map(|fileno| unsafe { libc::dup(fileno) })
+            .map(|fileno| unsafe { libc::fcntl(fileno, libc::F_DUPFD_CLOEXEC, 0) })
```

Comment:
Can you leave a comment explaining a bit more what this does and why it is needed?